cmd/tools/grafana/grafana.go

```diff
@@ -247,153 +338,87 @@ func doImport(_ *cobra.Command, _ []string) {
 	}
 }
 
-func checkAndCreateFolder() {
-	var folderExists = doesGrafanaFolderExist()
-	var err error
-
-	// first folder is Harvest 2.0 and second folder is 7mode
-	for folder, value := range folderExists {
-		if value {
-			fmt.Printf("folder [%s] exists in Grafana - OK\n", folder.folderName)
-		} else if err = createFolder(opts, folder); err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		} else {
-			fmt.Printf("created Grafana folder [%s] - OK\n", folder.folderName)
-		}
+func validateImport() {
+	// default case
+	if opts.dir == "" && opts.serverfolder.name == "" {
+		opts.dir = "grafana/dashboards"
+		opts.dir = path.Join(homePath, opts.dir)
 	}
 
-}
-
-func doesGrafanaFolderExist() map[*Folder]bool {
-	folderExists := make(map[*Folder]bool)
-	var err error
-	if folderExists, err = checkFolder(opts); err != nil {
-		fmt.Println(err)
+	exitIfMissing(opts.dir, "directory")
+	files, err := ioutil.ReadDir(opts.dir)
+	if err != nil {
+		fmt.Printf("Error while reading dir [%s] is the directory correct?\n", opts.dir)
 		os.Exit(1)
 	}
-	return folderExists
-}
-
-func askForToken() {
-	// ask for API token if not provided as arg and validate
-	if err := checkToken(opts, false); err != nil {
-		fmt.Println(err)
+	if len(files) == 0 {
+		fmt.Printf("No dashboards found in [%s] is the directory correct?\n", opts.dir)
 		os.Exit(1)
 	}
 }
 
-func adjustOptions() {
-	opts.config = conf.ConfigPath(opts.config)
-	homePath = conf.GetHarvestHomePath()
-	if opts.dircDOT != "" {
-		opts.dircDOT = path.Join(homePath, opts.dircDOT)
-		exitIfMissing(opts.dircDOT, "directory-cdot")
-	}
-	if opts.dir7mode != "" {
-		opts.dir7mode = path.Join(homePath, opts.dir7mode)
-		exitIfMissing(opts.dir7mode, "directory-7mode")
-	}
-	if opts.dircDOT == "" && opts.dir7mode == "" {
-		opts.dir = path.Join(homePath, opts.dir)
-		exitIfMissing(opts.dir, "directory")
-		opts.dircDOT = path.Join(opts.dir, "/cmode")
-		opts.dir7mode = path.Join(opts.dir, "/7mode")
+func initImportVars() {
+	opts.dirGrafanaFolderMap = make(map[string]*Folder)
+
+	m := make(map[string]*Folder)
+
+	// default behaviour
+	if opts.dir == "grafana/dashboards" && opts.serverfolder.name == "" {
+		m[path.Join(opts.dir, "/cmode")] = &Folder{name: "Harvest-" + harvestRelease + "-cDOT"}
+		m[path.Join(opts.dir, "/7mode")] = &Folder{name: "Harvest-" + harvestRelease + "-7mode"}
+	} else if opts.dir != "" && opts.serverfolder.name != "" {
```

Comment:
is there a missing case here? what if opts.dir != "" and  opts.serverfolder.name == "" or did some earlier check disallow that?

Reply:
yes that will be disallowed with cobra manadatory check. Both serverfolder/directory or none.